core/src/main/java/org/pentaho/platform/dataaccess/datasource/api/DatasourceService.java

```diff
@@ -154,4 +156,48 @@ public static boolean isDSWDatasource( Domain domain ) {
     }
     return !isMetadataDatasource( domain );
   }
+
+  protected String isMetaDataSource( String id ) {
```

Comment:
normally when functions start with "is" they answer a question and the return value is boolean. just in like in `DatasourceService#isMetadataDatasource(domain)`

This function is return a string, the id of the metadatasource, so the method signature should get something like
`getMetadataId(id)` or `getIdgetMetadata(id)`